ambari-server/src/main/java/org/apache/ambari/server/orm/entities/UserAuthenticationEntity.java

```diff
@@ -101,13 +101,35 @@ public void setAuthenticationType(UserAuthenticationType authenticationType) {
   }
 
   public String getAuthenticationKey() {
+    int firstCommaIndex = authenticationKey.indexOf(",");
+    if (firstCommaIndex != -1) {
+      return authenticationKey.substring(0, firstCommaIndex);
+    }
+    return authenticationKey;
+  }
+
+  public String getFullAuthenticationKey() {
     return authenticationKey;
   }
 
   public void setAuthenticationKey(String authenticationKey) {
     this.authenticationKey = authenticationKey;
   }
 
+  public void updateAuthenticationKey(String newAuthenticationKey, int historyCount) {
+    int currentCount = this.authenticationKey.split(",").length;
+    while(currentCount >= historyCount){
+      int lastCommaIndex = this.authenticationKey.lastIndexOf(",");
+      if(lastCommaIndex == -1){
+        this.authenticationKey = "";
+      } else {
+        this.authenticationKey = this.authenticationKey.substring(0, lastCommaIndex);
+      }
+      currentCount--;
+    }
+    this.authenticationKey = newAuthenticationKey + "," + this.authenticationKey;
+  }
```

Comment:
@himanshumaurya09876 An while loop is not very elegant or readable.
```suggestion
  public void updateAuthenticationKey(String newAuthenticationKey, int historyCount) {
    if (this.authenticationKey == null || this.authenticationKey.isEmpty()) {
        this.authenticationKey = newAuthenticationKey;
    } else {
        String[] keys = this.authenticationKey.split(",");
        List<String> keyList = new ArrayList<>(Arrays.asList(keys));
        if (keyList.size() >= historyCount) {
            keyList = keyList.subList(0, historyCount - 1);
        }
        keyList.add(0, newAuthenticationKey);
        this.authenticationKey = String.join(",", keyList);
    }
}
```